### ISSUE TITLE
Accessibility: Modal keyboard focus not trapped for screen reader users with arrow keys

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51926]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The modal has a correct keyboard trap when using the tab key, but it does not have a keyboard trap for screen reader users using the arrow keys.  Screen reader users should not be able to leave the modal without closing it.

![Image](https://github.com/user-attachments/assets/ebd4aa07-d984-4c89-9ed9-5a15044066b6)

### Component or Page: 

Page: Product Single (Simple)  

### Name affected element:

Product zoom modal

### Relevant code snippets:

`<div class=""pswp__ui pswp__ui--one-slide pswp__ui--fit pswp__ui--over-close"">
			<div class=""pswp__top-bar"">
				<div class=""pswp__counter"">1 / 1</div>
				<button class=""pswp__button pswp__button--close"" aria-label=""Close (Esc)""></button>
				<button class=""pswp__button pswp__button--share pswp__element--disabled"" aria-label=""Share""></button>
				<button class=""pswp__button pswp__button--fs"" aria-label=""Toggle fullscreen""></button>
				<button class=""pswp__button pswp__button--zoom"" aria-label=""Zoom in/out""></button>
				<div class=""pswp__preloader"">...</div>`

### Recommended fix:

Add aria-modal=""true"" and role=""dialog"" to the modal container to help assistive technology communicate the bounds of the modal and distinguish it from the rest of the page's content.
 
 There should also be an aria-label or aria-labelledby attribute that names the modal.
 
 Learn more about aria-modal at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal and the dialog role at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role.
